sklearn/ensemble/tests/test_bagging.py

```diff
@@ -583,19 +603,53 @@ def predict(self, X):
         return self.classes_[np.zeros(X.shape[0], dtype=int)]
 
 
-def test_bagging_sample_weight_unsupported_but_passed():
-    estimator = BaggingClassifier(DummyZeroEstimator())
+@pytest.mark.parametrize("estimator_cls", [BaggingClassifier, BaggingRegressor])
+def test_bagging_sample_weight_unsupported_but_passed(estimator_cls):
+    estimator = estimator_cls(DummyZeroEstimator())
     rng = check_random_state(0)
 
     estimator.fit(iris.data, iris.target).predict(iris.data)
-    with pytest.raises(ValueError):
+    msg = (
+        "fit got unexpected argument(s) {'sample_weight'}, which are not requested "
+        "metadata in any object."
+    )
+    with pytest.raises(TypeError, match=re.escape(msg)):
         estimator.fit(
             iris.data,
             iris.target,
             sample_weight=rng.randint(10, size=(iris.data.shape[0])),
         )
 
 
+@pytest.mark.parametrize("estimator_cls", [BaggingClassifier, BaggingRegressor])
+@pytest.mark.parametrize("fit_params_type", ["list", "array"])
+@pytest.mark.parametrize("use_sample_weight", [True, False])
+def test_bagging_with_arbitrary_fit_params(
```

Comment:
isn't this test covered in `test_metaestimators_metadata_routing.py`?

Reply:
I added these tests because they did catch an error that the tests in `test_metaestimators_metadata_routing.py` didn't. I cannot remember what exactly the error was, but below is a change that will make these tests fail whereas `test_metaestimators_metadata_routing.py` pass. Change this line:

https://github.com/BenjaminBossan/scikit-learn/blob/384127292b610785030ddd6fad0e6b1ee4376f4a/sklearn/ensemble/_bagging.py#L498

like this:

```python
-                fit_params=routed_params.base_estimator.fit,
+                fit_params={"sample_weight": None},
```

Maybe instead of adding these tests, we have to fix `test_metaestimators_metadata_routing.py`, but I'm not sure how.